version: '3.9'
services:
  pg-0:
    image: docker.io/bitnami/postgresql-repmgr:latest
    container_name: pg-0
    restart: on-failure
    ports:
      - 5432
    volumes:
      - pg_0_data:/bitnami/postgresql
    env_file:
      - ./conf/pg-0.env

  pg-1:
    image: docker.io/bitnami/postgresql-repmgr:latest
    container_name: pg-1
    restart: on-failure
    ports:
      - 5432
    volumes:
      - pg_1_data:/bitnami/postgresql
    env_file:
      - ./conf/pg-1.env
    depends_on:
      - pg-0

  pg-2:
    image: docker.io/bitnami/postgresql-repmgr:latest
    container_name: pg-2
    restart: on-failure
    ports:
      - 5432
    volumes:
      - pg_2_data:/bitnami/postgresql
    env_file:
      - ./conf/pg-2.env
    depends_on:
      - pg-0

  pgpool:
    image: docker.io/bitnami/pgpool:latest
    container_name: pgpool
    restart: on-failure
    ports:
      - 5432:5432
    env_file:
      - ./conf/pgpool.env
    healthcheck:
      test: ["CMD", "/opt/bitnami/scripts/pgpool/healthcheck.sh"]
      interval: 10s
      timeout: 5s
      retries: 5
    depends_on:
      - pg-0
      - pg-1
      - pg-2

  py_script:
    build: ./scripts
    container_name: py_script
    restart: on-failure
    depends_on:
      - pgpool
      - pg-0
      - pg-1
      - pg-2

  pgadmin:
    image: dpage/pgadmin4
    container_name: pgadmin
    restart: always
    volumes:
      - pgadmin-data:/var/lib/pgadmin
      - ./conf/pgadmin/servers.json:/pgadmin4/servers.json
    env_file:
      - ./conf/pgadmin.env
    ports:
      - 5050:80

volumes:
  pg_0_data:
    driver: local
  pg_1_data:
    driver: local
  pg_2_data:
    driver: local
  pgadmin-data:
    driver: local